src/Interpreters/ClusterDiscovery.cpp

```diff
@@ -123,23 +124,60 @@ ClusterDiscovery::ClusterDiscovery(
     Poco::Util::AbstractConfiguration::Keys config_keys;
     config.keys(config_prefix, config_keys);
 
+    multicluster_discovery_paths = std::make_shared<std::vector<std::shared_ptr<MulticlusterDiscovery>>>();
+
     for (const auto & key : config_keys)
     {
         String cluster_config_prefix = config_prefix + "." + key + ".discovery";
         if (!config.has(cluster_config_prefix))
             continue;
 
-        String zk_name_and_root = config.getString(cluster_config_prefix + ".path");
-        if (zk_name_and_root.empty())
-            throw Exception(ErrorCodes::NO_ELEMENTS_IN_CONFIG, "ZooKeeper path for cluster '{}' is empty", key);
-        String zk_root = zkutil::extractZooKeeperPath(zk_name_and_root, true);
-        String zk_name = zkutil::extractZooKeeperName(zk_name_and_root);
+        String zk_name_and_root = config.getString(cluster_config_prefix + ".path", "");
+        String zk_multicluster_name_and_root = config.getString(cluster_config_prefix + ".multicluster_root_path", "");
```

Comment:
The option names `multicluster_root_path` and `path` seem a bit inconsistent to me. I would expect both to be either `path`/`multicluster_path` or `root_path`/`multicluster_root_path`. Or is it that `multicluster_root_path` corresponds to multiple clusters, meaning we would need several `<path>` entries to define them? If so, probably makes sense.

Reply:
And could you also update documentation https://github.com/ClickHouse/ClickHouse/blob/master/docs/en/operations/cluster-discovery.md accordingly with the new feature, please?